src/transport_ofi.c

```diff
@@ -1157,20 +1158,10 @@ int query_for_fabric(struct fabric_info *info)
     hints.addr_format         = FI_FORMAT_UNSPEC;
     domain_attr.data_progress = FI_PROGRESS_AUTO;
```

Comment:
Just a note that #1059 possibly motivates a guard warning about and/or preventing the usage of `FI_MR_ENDPOINT` (and thus `FI_PROGRESS_MANUAL`) except for providers where it makes sense.  But come to think of it, maybe this PR already accomplishes that?